File: src/main/scala/io/snyk/plugin/ui/settings/SnykProjectSettingsStateService.scala
package io.snyk.plugin.ui.settings

import java.util.Objects.isNull

import com.intellij.openapi.components.{PersistentStateComponent, ServiceManager, State, Storage}
import com.intellij.openapi.project.Project
import com.intellij.util.xmlb.XmlSerializerUtil

@State(
  name = "SnykProjectSettingsState",
  storages = Array(new Storage("snyk.project.settings.xml"))
)
class SnykProjectSettingsStateService
  extends PersistentStateComponent[SnykProjectSettingsStateService] {

  private var additionalParameters = ""

  override def getState: SnykProjectSettingsStateService = this

  override def loadState(state: SnykProjectSettingsStateService): Unit = {
    XmlSerializerUtil.copyBean(state, this)
  }

  def getAdditionalParameters: String = additionalParameters

  def setAdditionalParameters(newParameters: String): Unit = this.additionalParameters = newParameters
}

object SnykProjectSettingsStateService {
  def getInstance(project: Project): Option[SnykProjectSettingsStateService] = {
    if (isNull(project) || project.isDefault) {
      Option.empty
    } else {
      Option(ServiceManager.getService(project, classOf[SnykProjectSettingsStateService]))
    }
  }
}

File: src/main/kotlin/io/snyk/plugin/settings/SnykProjectSettingsStateService.kt
package io.snyk.plugin.settings

import com.intellij.openapi.components.*
import com.intellij.util.xmlb.XmlSerializerUtil

@Service
@State(
    name = "SnykProjectSettingsState",
    storages = [Storage("snyk.project.settings.xml", roamingType = RoamingType.DISABLED)]
)
class SnykProjectSettingsStateService : PersistentStateComponent<SnykProjectSettingsStateService> {

    private var additionalParameters = ""

    override fun getState(): SnykProjectSettingsStateService = this

    override fun loadState(state: SnykProjectSettingsStateService) {
        XmlSerializerUtil.copyBean(state, this)
    }

    fun getAdditionalParameters(): String = additionalParameters

    fun setAdditionalParameters(newParameters: String) {
        this.additionalParameters = newParameters
    }
}
